### PR TITLE
1850 authenticate eventstore subscription requests

### LINKED ISSUE #1850: Authenticate eventstore subscription requests

### PR DESCRIPTION
**Pull request by alexopenline**
## Proposed changes

authenticate eventstore subscription requests

## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

### CODE REVIEW

File: packages/server/events-processing-platform/.env [hunk 1]
```diff
@@ -1,6 +1,17 @@
 SERVICE_NAME: events-processing-platform
 
-EVENT_STORE_CONNECTION_STRING: esdb://localhost:2113?tls=false
+EVENT_STORE_CONNECTION_STRING: esdb://localhost:2113
+EVENT_STORE_CONNECTION_TLS_DISABLED: true
+EVENT_STORE_CONNECTION_TLS_VERIFY_CERT: false
+EVENT_STORE_CONNECTION_KEEP_ALIVE_TIMEOUT: 10000
+EVENT_STORE_CONNECTION_KEEP_ALIVE_INTERVAL: 10000
+EVENT_STORE_CONNECTION_USERNAME:
```
Comment (tsearle) [inline packages/server/events-processing-platform/.env:8]: These new env vars need to also be added in 
deployment/infra/k8s/customer-os-api.yaml

File: packages/server/events-processing-platform/eventstroredb/config.go [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
 
 type EventStoreConfig struct {
-	ConnectionString string `env:"EVENT_STORE_CONNECTION_STRING" validate:"required"`
+	ConnectionString   string `env:"EVENT_STORE_CONNECTION_STRING" validate:"required"`
+	TlsDisable         bool   `env:"EVENT_STORE_CONNECTION_TLS_DISABLED" envDefault:"false"`
+	TlsVerifyCert      bool   `env:"EVENT_STORE_CONNECTION_TLS_VERIFY_CERT" envDefault:"true"`
+	KeepAliveTimeout   int    `env:"EVENT_STORE_CONNECTION_KEEP_ALIVE_TIMEOUT_MS" envDefault:"10000"`
+	KeepAliveInterval  int    `env:"EVENT_STORE_CONNECTION_KEEP_ALIVE_INTERVAL_MS" envDefault:"10000"`
+	ConnectionUser     string `env:"EVENT_STORE_CONNECTION_USERNAME"`
+	ConnectionPassword string `env:"EVENT_STORE_CONNECTION_PASSWORD"`
```
Comment (tsearle) [inline packages/server/events-processing-platform/eventstroredb/config.go]: add ",unset"
for passwords

File: packages/server/events-processing-platform/eventstroredb/config.go [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+
+	AdminUsername      string `env:"EVENT_STORE_ADMIN_USERNAME"`
+	AdminPassword      string `env:"EVENT_STORE_ADMIN_PASSWORD"`
+	OperationsUsername string `env:"EVENT_STORE_OPS_USERNAME"`
+	OperationsPassword string `env:"EVENT_STORE_OPS_PASSWORD"`
```
Comment (tsearle) [inline packages/server/events-processing-platform/eventstroredb/config.go]: add ",unset" for passwords

File: packages/server/events-processing-platform/eventstroredb/config.go [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
+	ConnectionString   string `env:"EVENT_STORE_CONNECTION_STRING" validate:"required"`
+	TlsDisable         bool   `env:"EVENT_STORE_CONNECTION_TLS_DISABLED" envDefault:"false"`
+	TlsVerifyCert      bool   `env:"EVENT_STORE_CONNECTION_TLS_VERIFY_CERT" envDefault:"true"`
+	KeepAliveTimeout   int    `env:"EVENT_STORE_CONNECTION_KEEP_ALIVE_TIMEOUT_MS" envDefault:"10000"`
+	KeepAliveInterval  int    `env:"EVENT_STORE_CONNECTION_KEEP_ALIVE_INTERVAL_MS" envDefault:"10000"`
+	ConnectionUser     string `env:"EVENT_STORE_CONNECTION_USERNAME"`
+	ConnectionPassword string `env:"EVENT_STORE_CONNECTION_PASSWORD"`
+
+	AdminUsername      string `env:"EVENT_STORE_ADMIN_USERNAME"`
+	AdminPassword      string `env:"EVENT_STORE_ADMIN_PASSWORD"`
```
Comment (tsearle) [inline packages/server/events-processing-platform/eventstroredb/config.go]: add ",unset" for passwords

File: packages/server/events-processing-platform/.env (hunk 1 above)
Comment (alexopenline) [reply packages/server/events-processing-platform/.env:8]: these are events-platform vars.
whole app to be added.
will create a task

File: packages/server/events-processing-platform/eventstroredb/config.go (hunk 2 above)
Comment (alexopenline) [reply packages/server/events-processing-platform/eventstroredb/config.go]: done

File: packages/server/events-processing-platform/eventstroredb/config.go (hunk 3 above)
Comment (alexopenline) [reply packages/server/events-processing-platform/eventstroredb/config.go]: done

File: packages/server/events-processing-platform/.env (hunk 1 above)
Comment (tsearle) [reply packages/server/events-processing-platform/.env:8]: the app has already been added by silviu

**Review by tsearle — APPROVED** [review]

**Pull request merged**